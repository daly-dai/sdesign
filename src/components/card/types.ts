import { CardProps } from 'antd';
import { ReactNode } from 'react';

export interface SCardProps extends Omit<CardProps, 'children'> {
  children?: ReactNode;
  hasBottomPadding?: boolean; // 底部是否包含边距
}
