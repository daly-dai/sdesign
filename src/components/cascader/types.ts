import { CascaderProps } from 'antd';

export type SingleValueType = (string | number)[];
export type ValueType = SingleValueType | SingleValueType[] | undefined;

export type SCascaderProps = Omit<
  CascaderProps<any>,
  'onChange' | 'value' | 'defaultValue' | 'multiple'
> & {
  onChange?: (value: string, selectOptions?: any[]) => void;
  value?: string | null | ValueType;
  options?: any[];
  defaultValue?: string | ValueType;
  multiple?: boolean;
};
