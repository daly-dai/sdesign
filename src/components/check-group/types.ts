import { Checkbox } from 'antd';
import { ComponentProps, HTMLAttributes, ReactNode } from 'react';

import { ExtraComType } from '@dalydb/sdesign/types/base';

export type CheckboxGroupType = HTMLAttributes<object> &
  Omit<ComponentProps<typeof Checkbox.Group>, 'onChange' | 'value'>;
export type CheckboxValueType = any;

export type SCheckGroupProps = CheckboxGroupType &
  ExtraComType & {
    value?: CheckboxValueType[] | undefined | string;
    onChange?: (value: string | undefined) => void;
    children?: ReactNode;
  };
