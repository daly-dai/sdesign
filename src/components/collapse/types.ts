import { ButtonProps } from 'antd/es/button/button';

export interface SCollapseProps extends ButtonProps {
  collapse?: boolean;
  setCollapse?: (collapse: boolean) => void;
  onExpand?: (collapse: boolean) => void;
}
