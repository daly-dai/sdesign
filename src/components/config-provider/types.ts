import { ReactNode } from 'react';

export type SConfigProviderType = {
  // 全部的字典
  globalDict?: Record<string, any>;
  // 上传组件接口地址地址
  uploadUrl?: string;
  children?: ReactNode;
  prefixCls?: string;
};

export interface ConfigContextProps {
  globalDict?: Record<string, Record<string, any> | any>;
  uploadUrl?: string;
  getPrefixCls: (suffixCls?: string, customizePrefixCls?: string) => string;
}
