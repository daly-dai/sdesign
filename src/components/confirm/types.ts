import { ButtonProps, PopconfirmProps } from 'antd';

import { tuple } from '@dalydb/sdesign/utils/types';

const ConfirmTypes = tuple('pop', 'modal');
export type ConfirmType = (typeof ConfirmTypes)[number];

export interface SConfirmProps {
  onConfirm?: () => void; // 确认的回调函数
  onCancel?: () => void; // 取消的回调函数
  type?: ConfirmType;
  text?: string;
  title?: React.ReactNode;
  description?: React.ReactNode;
  children?: React.ReactNode;
  popConfirmProps?: PopconfirmProps; // 弹出框的属性传参
  buttonProps?: ButtonProps; // 按钮的属性传参
  disabled?: boolean;
}
