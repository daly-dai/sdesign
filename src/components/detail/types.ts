import { DescriptionsProps } from 'antd';
import { DescriptionsItemType } from 'antd/es/descriptions';
import React, { ReactNode } from 'react';

import { FileListProps, PageTitleProps } from '@daly/sdesign';
import { tuple } from '@daly/sdesign/utils/types';

const ItemTypes = tuple(
  'text',
  'empty',
  'file',
  'dict',
  'img',
  'rangeTime',
  'checkbox',
  'placeholder',
);

export type ItemType = (typeof ItemTypes)[number];

export interface DictReflect {
  label?: string;
  name?: string;
}

export type SDetailItemType = DetailItemType & {
  label?: ReactNode;
  name?: string | string[];
  render?: (value?: any, dataSource?: any) => ReactNode;
  span?: number;
  /** * @desc 隐藏该属性 */
  hidden?: boolean;
};

export type DetailItemType = {
  type?: ItemType;
  fileProps?: Partial<FileListProps>;
  dictReflect?: DictReflect;
  dictKey?: string;
  value?: any;
  dictMap?: Record<string, string> | any[] | null;
  render?: (value?: any, dataSource?: any) => ReactNode;
  dataSource?: Record<string, any>;
};

export type SDetailItem = SDetailItemType &
  Omit<DescriptionsItemType, 'children' | 'key'>;

export interface SDetailProps
  extends Omit<DescriptionsProps, 'items' | 'title'> {
  titleDesc?: ReactNode;
  titleAction?: ReactNode;
  dataSource?: Record<string, any>;
  items?: SDetailItem[];
  labelStyle?: React.CSSProperties;
  contentStyle?: React.CSSProperties;
  hasCardBg?: boolean;
  container?: React.ComponentType<any>;
  title?: string | ReactNode;
  /** @description dataSource中的key */
  detailName?: string;
}

export interface SDetailGroupItem {
  groupTitle?: string | ReactNode;
  groupTitleProps?: Omit<PageTitleProps, 'title'>;
  groupContainer?: React.ComponentType<any>;
  groupItems?: SDetailProps[];
  items?: SDetailItem[];
  itemProps?: Omit<SDetailProps, 'items' | 'dataSource'>;
  dataSource?: Record<string, any>;
  /*** @desc 隐藏该属性*/
  hidden?: boolean;
}

export interface SDetailGroupProps {
  items?: SDetailGroupItem[];
  dataSource?: Record<string, any>;
}
