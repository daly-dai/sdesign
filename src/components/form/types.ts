import {
  Cascader,
  Checkbox,
  ColProps,
  DatePicker,
  FormItemProps,
  FormProps,
  Input,
  InputNumber,
  Radio,
  RowProps,
  Slider,
  Switch,
  Table,
  TimePicker,
  TreeSelect,
  Upload,
} from 'antd';
import { NamePath } from 'antd/es/form/interface';
import React, { ComponentProps, HTMLAttributes, ReactNode } from 'react';

import SCascader from '../cascader';
import SCheckGroup from '../check-group';
import SDatePicker from '../date-picker';
import SDatePickerRange from '../date-picker-range';
import SDependency from '../dependency';
import { RenderChildren } from '../dependency/types';
import SRadioGroup from '../radio-group';
import SSelect from '../select';
import SUpload from '../upload';
import InternalUpload from '../upload/instance';

import { RegKeyType } from '@dalydb/sdesign/types/reg';

export type FormFieldMapType = {
  input: typeof Input;
  inputNumber: typeof InputNumber;
  password: typeof Input.Password;
  textarea: typeof Input.TextArea;
  select: typeof SSelect;
  slider: typeof Slider;
  radio: typeof Radio;
  radioGroup: typeof SRadioGroup;
  switch: typeof Switch;
  treeSelect: typeof TreeSelect;
  upload: typeof Upload;
  SUpload: typeof InternalUpload;
  SUploadDragger: typeof SUpload.Dragger;
  SUploadPicture: typeof SUpload.Picture;
  datePicker: typeof DatePicker;
  SDatePicker: typeof SDatePicker;
  datePickerRange: typeof DatePicker.RangePicker;
  SDatePickerRange: typeof SDatePickerRange;
  timePicker: typeof TimePicker;
  timePickerRange: typeof TimePicker.RangePicker;
  checkbox: typeof Checkbox;
  checkGroup: typeof SCheckGroup;
  cascader: typeof Cascader;
  SCascader: typeof SCascader;
  table: typeof Table;
  dependency: typeof SDependency;
};

export type FormComType = keyof FormFieldMapType;

export type FormComPropsType = HTMLAttributes<object> &
  ComponentProps<FormFieldMapType[FormComType]>;

export type FormItemType = FormComType | 'placeholder';
export interface ItemsProps
  extends Omit<FormItemProps, 'label | name' | 'required'> {
  label?: ReactNode;
  name?: NamePath;
  style?: React.CSSProperties;
  // 组件的类型
  type?: FormItemType;
  // 依赖的字段，只在type为dependency时生效
  depNames?: string[];
  // 组件的配置项
  fieldProps?: FormComPropsType;
  // 自定义组件
  customCom?: ReactNode | RenderChildren<any>;
  // 校验规则
  regKey?: RegKeyType;
  // 自定义渲染
  render?: RenderChildren<any>;
  // 是否为必填
  required?: string | boolean;
  // 只读模式
  readonly?: boolean;
  // 嵌套的数据结构
  formName?: string;
  children?: ReactNode;
}

export interface SFormItems extends ItemsProps {
  colProps?: ColProps;
  hidden?: boolean;
}

export interface SFormProps extends FormProps {
  rowProps?: RowProps;
  children?: ReactNode;
  items?: SFormItems[];
  columns?: number;
  // 是否为必填,提示文字
  required?: string | boolean;
  onFinish?: (e: any) => void;
  onReset?: (e: any) => void;
  // 只读模式
  readonly?: boolean;
  // 嵌套的数据结构
  formName?: string;
}

export type GroupItemsType = {
  container?: React.ComponentType<any>;
  columns?: number;
  title?: ReactNode;
  items?: SFormItems[];
  rowProps?: RowProps;
  // 嵌套的数据结构
  formName?: string;
};

// 分组表单
export interface SFormGroupProps extends FormProps {
  groupItems?: GroupItemsType[];
  onFinish?: (e: any) => void;
  onReset?: (e: any) => void;
  container?: React.ComponentType<any>;
  // 嵌套的数据结构
  formName?: string;
  children?: ReactNode;
  /** @description  只读模式 */
  readonly?: boolean;
}

export interface SearchProps extends SFormProps {
  // 默认是否展开
  defaultExpand?: boolean;
  showExpand?: boolean;
  // 展开收起的回调函数
  onExpand?: (expand: boolean) => void;
  // 展开的行数
  expandLine?: number;
  actionNode?: ReactNode;
  /** @description 自定义组件容器 */
  container?: React.ComponentType<any>;
  isCard?: boolean;
}
