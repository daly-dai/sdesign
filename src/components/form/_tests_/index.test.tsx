import { fireEvent, render } from '@testing-library/react';
import { Button, Form, FormInstance } from 'antd';
import React, { useEffect } from 'react';
import { describe, expect, it } from 'vitest';

import { default as SForm, default as SFrom } from '../index';
import { SFormItems } from '../types';

import mountTest from '@dalydb/sdesign/tests/mountTest';
import rtlTest from '@dalydb/sdesign/tests/rtlTest';
import { waitFakeTimer } from '@dalydb/sdesign/tests/utils';

describe('SFrom', () => {
  mountTest(SFrom);
  mountTest(SFrom.Item);
  mountTest(SFrom.Group);
  mountTest(SFrom.Search);

  rtlTest(SFrom);
  rtlTest(SFrom.Item);
  rtlTest(SFrom.Group);
  rtlTest(SFrom.Search);

  // formName 嵌套数据类型,items表单配置,columns列数,readonly只读

  const items: SFormItems[] = [
    {
      label: '姓名',
      name: 'name',
    },
    {
      label: '年龄',
      name: 'age',
    },
  ];

  it('should support Items config', () => {
    const { container } = render(<SForm items={items}></SForm>);

    expect(container).toMatchSnapshot();
  });

  it('should support formName', () => {
    let formData = {};

    const Demo = () => {
      const [form] = Form.useForm();

      const handleClick = () => {
        const data = form.getFieldsValue();
        formData = data;
      };
      return (
        <>
          <SForm formName="test" items={items} form={form} />
          <Button type="primary" id="formSubmit" onClick={handleClick}>
            确 认
          </Button>
        </>
      );
    };

    const { getByText } = render(<Demo />);

    waitFakeTimer(300);

    const submit = getByText('确 认');
    fireEvent.click(submit);

    expect(formData).toEqual({
      test: {
        name: undefined,
        age: undefined,
      },
    });
  });

  it('should support setDefaultData by formName', () => {
    let formData = {};
    let formInstance: FormInstance<any> | null = null;

    const Demo = () => {
      const [form] = Form.useForm();

      useEffect(() => {
        formInstance = form;
      }, []);

      const handleClick = () => {
        const data = form.getFieldsValue();
        formData = data;
      };
      return (
        <>
          <SForm formName="test" items={items} form={form} />
          <Button type="primary" id="formSubmit" onClick={handleClick}>
            确 认
          </Button>
        </>
      );
    };

    const { getByText } = render(<Demo />);

    waitFakeTimer(300);

    // @ts-ignore
    formInstance?.setFieldsValue?.({
      test: {
        name: 'test',
        age: 18,
      },
    });

    const submit = getByText('确 认');
    fireEvent.click(submit);

    expect(formData).toEqual({
      test: {
        name: 'test',
        age: 18,
      },
    });
  });

  it('should support columns default 1', () => {
    const { container } = render(<SForm items={items} />);
    const columnsItems = container.querySelectorAll('.ant-col-24');

    expect(columnsItems.length).toBe(2);
  });

  it('should support custom columns', () => {
    const { container } = render(<SForm columns={2} items={items} />);
    const columnsItems = container.querySelectorAll('.ant-col-12');

    expect(columnsItems.length).toBe(2);
  });

  it('should support children', () => {
    const { container } = render(
      <SForm>
        <SForm.Item label="姓名" name="name">
          <input />
        </SForm.Item>
      </SForm>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should support readonly', () => {
    const { container } = render(
      <SForm readonly>
        <SForm.Item label="姓名" name="name">
          <input />
        </SForm.Item>
      </SForm>,
    );

    expect(container).toMatchSnapshot();
  });

  it('readonly width disabled', () => {
    const { container } = render(<SForm readonly items={items}></SForm>);

    const columnsItems = container.querySelectorAll('.ant-input-disabled');

    expect(columnsItems.length).toBe(2);
  });
});
