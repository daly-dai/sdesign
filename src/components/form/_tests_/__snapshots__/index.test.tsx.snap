// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SFrom > rtl render > component should be rendered correctly in RTL direction 1`] = `
<form
  class="ant-form ant-form-vertical ant-form-rtl css-dev-only-do-not-override-3rel02 sd-form"
>
  <div
    class="ant-row ant-row-rtl css-dev-only-do-not-override-3rel02"
    style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
  />
</form>
`;

exports[`SFrom > rtl render > component should be rendered correctly in RTL direction 2`] = `
<div
  class="ant-form-item css-dev-only-do-not-override-3rel02"
>
  <div
    class="ant-row ant-row-rtl ant-form-item-row css-dev-only-do-not-override-3rel02"
  >
    <div
      class="ant-col ant-form-item-control ant-col-rtl css-dev-only-do-not-override-3rel02"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <span
            class="ant-input-affix-wrapper ant-input-affix-wrapper-rtl css-dev-only-do-not-override-3rel02 ant-input-outlined"
          >
            <input
              class="ant-input ant-input-rtl css-dev-only-do-not-override-3rel02"
              placeholder="请输入"
              type="text"
              value=""
            />
            <span
              class="ant-input-suffix"
            >
              <span
                class="ant-input-clear-icon ant-input-clear-icon-hidden"
                role="button"
                tabindex="-1"
              >
                <span
                  aria-label="close-circle"
                  class="anticon anticon-close-circle"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="close-circle"
                    fill="currentColor"
                    fill-rule="evenodd"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SFrom > rtl render > component should be rendered correctly in RTL direction 3`] = `
<form
  class="ant-form ant-form-vertical ant-form-rtl css-dev-only-do-not-override-3rel02"
/>
`;

exports[`SFrom > rtl render > component should be rendered correctly in RTL direction 4`] = `
<div
  class="ant-card ant-card-bordered ant-card-rtl css-dev-only-do-not-override-3rel02"
  style="margin-bottom: 16px; padding: 16px 24px; border-radius: 8px;"
>
  <div
    class="ant-card-body"
    style="padding: 0px;"
  >
    <form
      class="ant-form ant-form-horizontal ant-form-rtl css-dev-only-do-not-override-3rel02"
    >
      <div
        class="ant-row ant-row-middle ant-row-rtl css-dev-only-do-not-override-3rel02"
        style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
      >
        <div
          class="ant-col ant-col-8 ant-col-offset-16 ant-col-rtl css-dev-only-do-not-override-3rel02"
          style="padding-left: 12px; padding-right: 12px;"
        >
          <div>
            <div
              class="acss-1bvc4cc"
            >
              <div
                class="ant-space css-dev-only-do-not-override-3rel02 ant-space-horizontal ant-space-rtl ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                style="column-gap: 12px;"
              >
                <div
                  class="ant-space-item"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default ant-btn-rtl"
                    type="reset"
                  >
                    <span>
                      重 置
                    </span>
                  </button>
                </div>
                <div
                  class="ant-space-item"
                >
                  <button
                    class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-primary ant-btn-rtl"
                    type="submit"
                  >
                    <span>
                      查 询
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`SFrom > should support Items config 1`] = `
<div>
  <form
    class="ant-form ant-form-vertical css-dev-only-do-not-override-3rel02 sd-form"
  >
    <div
      class="ant-row css-dev-only-do-not-override-3rel02"
      style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
    >
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-3rel02"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <div
          class="ant-form-item css-dev-only-do-not-override-3rel02"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
          >
            <div
              class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
            >
              <label
                class="ant-form-item-no-colon"
                for="name"
                title="姓名"
              >
                姓名
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-affix-wrapper css-dev-only-do-not-override-3rel02 ant-input-outlined"
                  >
                    <input
                      class="ant-input css-dev-only-do-not-override-3rel02"
                      id="name"
                      placeholder="请输入"
                      type="text"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        class="ant-input-clear-icon ant-input-clear-icon-hidden"
                        role="button"
                        tabindex="-1"
                      >
                        <span
                          aria-label="close-circle"
                          class="anticon anticon-close-circle"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="close-circle"
                            fill="currentColor"
                            fill-rule="evenodd"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                            />
                          </svg>
                        </span>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-3rel02"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <div
          class="ant-form-item css-dev-only-do-not-override-3rel02"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
          >
            <div
              class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
            >
              <label
                class="ant-form-item-no-colon"
                for="age"
                title="年龄"
              >
                年龄
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-affix-wrapper css-dev-only-do-not-override-3rel02 ant-input-outlined"
                  >
                    <input
                      class="ant-input css-dev-only-do-not-override-3rel02"
                      id="age"
                      placeholder="请输入"
                      type="text"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        class="ant-input-clear-icon ant-input-clear-icon-hidden"
                        role="button"
                        tabindex="-1"
                      >
                        <span
                          aria-label="close-circle"
                          class="anticon anticon-close-circle"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="close-circle"
                            fill="currentColor"
                            fill-rule="evenodd"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                            />
                          </svg>
                        </span>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`SFrom > should support children 1`] = `
<div>
  <form
    class="ant-form ant-form-vertical css-dev-only-do-not-override-3rel02 sd-form"
  >
    <div
      class="ant-row css-dev-only-do-not-override-3rel02"
      style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
    />
    <div
      class="ant-form-item css-dev-only-do-not-override-3rel02"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
      >
        <div
          class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
        >
          <label
            class="ant-form-item-no-colon"
            for="name"
            title="姓名"
          >
            姓名
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                id="name"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`SFrom > should support readonly 1`] = `
<div>
  <form
    class="ant-form ant-form-vertical css-dev-only-do-not-override-3rel02 sd-form"
  >
    <div
      class="ant-row css-dev-only-do-not-override-3rel02"
      style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
    />
    <div
      class="ant-form-item css-dev-only-do-not-override-3rel02"
    >
      <div
        class="ant-row ant-form-item-row css-dev-only-do-not-override-3rel02"
      >
        <div
          class="ant-col ant-form-item-label css-dev-only-do-not-override-3rel02"
        >
          <label
            class="ant-form-item-no-colon"
            for="name"
            title="姓名"
          >
            姓名
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control css-dev-only-do-not-override-3rel02"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                id="name"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
